app/src/lib/stores/repositories-store.ts

```diff
@@ -381,15 +395,26 @@ export class RepositoriesStore extends TypedBaseStore<
     // possible that the case differs (i.e we found `usera` but the actual login
     // is `userA`). In that case we want to update our database to persist the
     // login with the proper case.
-    if (existingOwner === undefined || existingOwner.login !== login) {
+    if (
+      existingOwner === undefined ||
+      existingOwner.login !== login ||
+      // This is added so that we update existing owners with an undefined type.
+      (ownerType !== undefined && existingOwner.type !== ownerType)
+    ) {
       id = existingOwner?.id
       const existingId = id !== undefined ? { id } : {}
-      id = await this.db.owners.put({ ...existingId, key, endpoint, login })
+      id = await this.db.owners.put({
+        ...existingId,
+        key,
+        endpoint,
+        login,
+        type: ownerType,
+      })
     } else {
       id = forceUnwrap('Missing owner id', existingOwner.id)
     }
 
-    return new Owner(login, endpoint, id)
+    return new Owner(login, endpoint, id, ownerType ?? existingOwner?.type)
```

Comment:
In the scenario of a skeleton repo being added because the repo details couldn't be retrieved, this will add an undefined owner type. The negative consequence is that if the owner was an organization type, we won't link to team references. But.. the likely hood that they could retrieve markdown to view for a repo and not the repo details would be slim.

Reply:
Yeah, the skeleton ones are always going to be funky. The skeleton repositories are super weird. This seems like a great tradeoff tho